flink-end-to-end-tests/flink-end-to-end-tests-hive/src/test/java/org/apache/flink/tests/hive/containers/HiveContainers.java

```diff
@@ -0,0 +1,248 @@
+            }
+        }
+
+        protected void setFilePermission(File file) {
+            file.setReadable(true, false);
+            file.setWritable(true, false);
+            file.setExecutable(true, false);
+        }
+
+        private void backupLogs() {
+            Path path =
+                    Paths.get(
+                            "/Users/ohmeatball/Desktop"); // DISTRIBUTION_LOG_BACKUP_DIRECTORY.get().orElse(null);
+            if (path == null) {
+                LOG.warn(
+                        "Property {} not set, logs will not be backed up.",
+                        DISTRIBUTION_LOG_BACKUP_DIRECTORY.getPropertyName());
+                return;
+            }
+            try {
+                Path dir =
+                        Files.createDirectory(
+                                Paths.get(
+                                        String.valueOf(path.toAbsolutePath()),
+                                        "hive-" + UUID.randomUUID()));
+
+                for (String filePathInDocker : getBackupLogPaths()) {
+                    copyFileFromContainer(
+                            filePathInDocker,
+                            dir.resolve(Paths.get(filePathInDocker).getFileName()).toString());
+                }
+            } catch (Throwable e) {
+                LOG.warn("Failed to backup logs...", e);
+            }
+        }
+
+        protected abstract List<String> getBackupLogPaths();
+
+        protected abstract void mount(List<String> initTableNames) throws IOException;
+
+        protected abstract int getNameNodeWebEndpointPort();
+    }
+
+    private static class Hive2Container extends HiveContainer {
+
+        private static final String METASTORE_LOG_PATH = "/tmp/hive/hive.log";
+        private static final String MYSQL_METASTORE_LOG_PATH = "/var/log/mysqld.log";
+        private static final String NAME_NODE_LOG_PATH =
+                "/var/log/hadoop-hdfs/hadoop-hdfs-namenode.log";
+        private static final int NAME_NODE_WEB_PORT = 50070;
+
+        public Hive2Container(List<String> initTableNames) {
+            super(DockerImageVersions.HIVE2, initTableNames);
+        }
+
+        @Override
+        protected List<String> getBackupLogPaths() {
+            // Detailed log paths are from
+            // https://github.com/prestodb/docker-images/tree/master/prestodb/hdp2.6-hive/files/etc/supervisord.d
+            return Arrays.asList(METASTORE_LOG_PATH, MYSQL_METASTORE_LOG_PATH, NAME_NODE_LOG_PATH);
+        }
+
+        @Override
+        protected void mount(List<String> initTableNames) {
+            withFileSystemBind(hiveWarehouseDir, hiveWarehouseDir);
+        }
+
+        @Override
+        protected int getNameNodeWebEndpointPort() {
+            return NAME_NODE_WEB_PORT;
+        }
+    }
+
+    private static class Hive3Container extends HiveContainer {
+
+        private static final String METASTORE_LOG_PATH = "/tmp/root/hive.log";
```

Comment:
Can these common properties be extract the parent class?

Reply:
no. Hive3 places logs in the "/tmp/root/hive.log", but hive2 puts logs in the "/tmp/hive/hive.log".